### PR TITLE
[CE-4000] Add support for Global HereNow

### PR DESCRIPTION
**Pull request by CraigLn**
Resolves HereNow response decoding when stateless or only querying a single channel

### CODE REVIEW

File: Tests/PubNubTests/Integration/OnboardingSnippets.swift [hunk 1]
```diff
@@ -85,30 +85,41 @@ class OnboardingSnippets: XCTestCase {
...
+        historyExpect.fulfill()
       }
-      historyExpect.fulfill()
     }
 
-    wait(for: [historyExpect], timeout: 10.0)
+    client.publish(channel: "pubnub_onboarding_channel",
+                   message: ["sender": configuration.uuid,
+                             "content": "Hello From SDK_NAME SDK"])
+    { result in
```
Comment (madisonsmith) [inline Tests/PubNubTests/Integration/OnboardingSnippets.swift]: Codacy Issue found: [Opening braces should be preceded by a single space and on the same line as the declaration.](https://app.codacy.com/app/PubNub/swift/pullRequest?prid=4143430)

**Comment by madisonsmith** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/PubNub/swift/pullRequest?prid=4143430&bid=14215654)

File: Tests/PubNubTests/Integration/OnboardingSnippets.swift (hunk 1 above)
Comment (madisonsmith) [inline Tests/PubNubTests/Integration/OnboardingSnippets.swift]: Codacy Issue found: [Unused parameter "result" in a closure should be replaced with _.](https://app.codacy.com/app/PubNub/swift/pullRequest?prid=4143430)

File: .pubnub.yml [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+      -
+        text: "Add support for Global HereNow"
+        type: bug
+      -
+        text: "Fixed HereNow stateless response decoding"
+        type: bug
+      -
+        text: "Fixed HereNow single channel response decoding"
+        type: bug
+    date: 2019-XX-XX
```
Comment (davidnub) [inline .pubnub.yml]: Should probably update the date here.

Comment (davidnub) [reply .pubnub.yml]: Oh are we doing this after release?

File: Tests/PubNubTests/Integration/PresenceEndpointIntegrationTests.swift [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+      }
+      hereNowExpect.fulfill()
+    }
+
+    wait(for: [hereNowExpect], timeout: 10.0)
+  }
+
+  // MARK: - Here Now Mutlti Channel
+
+  func testHereNow_MultieChannel_Stateless() {
```
Comment (davidnub) [inline Tests/PubNubTests/Integration/PresenceEndpointIntegrationTests.swift]: Typo: `MultieChannel` => `MultiChannel`

File: Tests/PubNubTests/Integration/PresenceEndpointIntegrationTests.swift [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    }
+    let token = client.add(listener)
+
+    client.subscribe(to: [channel], withPresence: true)
+
+    defer { token.cancel() }
+    wait(for: [hereNowExpect], timeout: 10.0)
+  }
+
+  func testHereNow_MultieChannel_State() {
```
Comment (davidnub) [inline Tests/PubNubTests/Integration/PresenceEndpointIntegrationTests.swift]: Typo: `MultieChannel` => `MultiChannel`

File: Tests/PubNubTests/Integration/PresenceEndpointIntegrationTests.swift [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    }
+    let token = client.add(listener)
+
+    client.subscribe(to: [channel], withPresence: true, setting: ["StateKey": "StateValue"])
+
+    defer { token.cancel() }
+    wait(for: [hereNowExpect], timeout: 10.0)
+  }
+
+  func testHereNow_MultieChannel_EmptyPresence() {
```
Comment (davidnub) [inline Tests/PubNubTests/Integration/PresenceEndpointIntegrationTests.swift]: Typo: `MultieChannel` => `MultiChannel`

File: .pubnub.yml (hunk 2 above)
Comment (CraigLn) [reply .pubnub.yml]: Just whenever we know the date of release.  Typically whenever we make the release PR (if there is one), after the release, or before if the date is predictable.

File: pubnub.yml [hunk 6]
```diff
@@ -0,0 +1 @@
+.pubnub.yml
```
[the remark this answers is not in the document]
Comment (CraigLn) [reply pubnub.yml]: You can't add hidden files to xcode, so I added the symlinked file to make it easier to do CRUD on the changelog

File: Tests/PubNubTests/Integration/PresenceEndpointIntegrationTests.swift (hunk 3 above)
Comment (CraigLn) [reply Tests/PubNubTests/Integration/PresenceEndpointIntegrationTests.swift]: Apparently I was unpacking my clothes around the time of writing these tests...

File: pubnub.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (davidnub) [reply pubnub.yml]: Oh ok.

**Pull request merged**